core/src/main/java/io/grpc/SynchronizationContext.java

```diff
@@ -113,13 +105,11 @@ public final void drain() {
    * #executeLater} in the lock, and call {@link #drain} outside the lock.
    */
   public final void executeLater(Runnable runnable) {
-    synchronized (lock) {
-      queue.add(checkNotNull(runnable, "runnable is null"));
-    }
+    queue.add(checkNotNull(runnable, "runnable is null"));
```

Comment:
This may change the current behavior because the ordering of the runnables may be changed. There seems no harm for the ordering change (May need confirm from @zhangkun83 ). This change seems better, because `executeLater` becomes always not blocking after the change.

Reply:
Discussed offline, the behavior is more or less the same. So no concern here.